proc-macros/src/render_client.rs

```diff
@@ -100,8 +100,30 @@ impl RpcDescription {
 			let params = method.params.iter().map(|(param, _param_type)| {
 				quote! { #serde_json::to_value(&#param)? }
 			});
-			quote! {
-				Some(vec![ #(#params),* ].into())
+			match method.param_kind {
+				ParamKind::Map => {
+					// Extract parameter names.
+					let param_names = extract_param_names(&method.signature.sig);
+					// Combine parameter names and values into tuples.
+					let params = param_names.iter().zip(params).map(|pair| {
+						let param = pair.0;
+						let value = pair.1;
+						quote! { (#param, #value) }
+					});
+					quote! {
+						Some(types::v2::ParamsSer::Map(
+								std::collections::BTreeMap::<&str, #serde_json::Value>::from(
+									[#(#params),*]
+									)
+								)
+							)
+					}
+				}
+				ParamKind::Array => {
+					quote! {
+						Some(vec![ #(#params),* ].into())
+					}
+				}
```

Comment:
Would be nice if this was DRY with lines 169..191.

Also would be nice if we allowed a flat `Box<RawValue>` variant here, but that's way beyond the scope of this PR :)